plugin/storage/grpc/factory.go

```diff
@@ -98,7 +107,8 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		for _, opt := range opts {
 			clientOpts = append(clientOpts, configgrpc.WithGrpcDialOption(opt))
 		}
-		return f.config.ToClientConn(context.Background(), componenttest.NewNopHost(), telset, clientOpts...)
+		noophost := NewNopHost()
```

Comment:
@chahatsagarmain instead of creating a nophost here, you can just pass `component.Host` into `NewFactoryWithConfig` from https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L115